### PR TITLE
Conrad crud

### PR DESCRIPTION
**Pull request by conrad-krueger**
Created the following:

**User Interface with attributes:** id (SERIAL PK), name (str), email (str) 

**- getAllUsers (get request)**
returns entire user table as a json.

**- getUserById (get request)**
returns row of users table on backend of user with the corresponding id parameter

**- deleteUser (delete request)**
delete row from user table (via id param)

**- createUser (post request)**
creates a user with a name and email value
409 error returned if email already exists

**All get, update, and delete** return 404 error if user is not found.
**All non-delete requests** return either 200 or 201 status code of the user if the request went through 
**If delete request is successful,** 200 status returned with "User deleted successfully"

### CODE REVIEW

File: server/src/routers/user.routes.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import express from "express";
+
+import {
+    getAllUsers,
+    getUserById,
+    updateUserName,
```
Comment (shaham-noorani) [inline server/src/routers/user.routes.ts]: realistically, a user should never update their name or email (esp this one bc they login with oauth)

File: server/src/controllers/user.controller.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        res.status(200).json({ message: "User deleted successfully" });
+    } catch (error: any) {
+        res.status(500).json({ message: error.message });
+    }
+};
+
+//Create Users
+export const createUser = async (req: Request, res: Response) => {
+    try {
+
```
Comment (shaham-noorani) [inline server/src/controllers/user.controller.ts:60]: go through and make sure formatting is consistent. I would recommend using something like prettier and having vs code "format on save"

**Review by shaham-noorani — CHANGES_REQUESTED** [review]
Just some small changes, the rest looks good!

File: server/src/routers/user.routes.ts (hunk 1 above)
Comment (conrad-krueger) [reply server/src/routers/user.routes.ts]: Deleted

File: server/src/controllers/user.controller.ts (hunk 2 above)
Comment (conrad-krueger) [reply server/src/controllers/user.controller.ts:60]: Used prettier

**Pull request merged**